File: examples/equos-nodejs-integration/src/create-character.ts
import { client } from "./client.js";

// Combine a brain (required) and an optional face/voice into a character.
// IDs come from the corresponding `create-*` scripts or from Equos Studio.
const brainId = process.env.EQUOS_BRAIN_ID;
if (!brainId) throw new Error("EQUOS_BRAIN_ID is not set");

const character = await client.characters.createCharacter({
  createEquosCharacterRequest: {
    name: "French Translator",
    livekitIdentity: `french-translator-${Date.now()}`,
    brainId,
    faceId: process.env.EQUOS_FACE_ID,
    voiceId: process.env.EQUOS_VOICE_ID,
    search: false,
  },
});

console.log("Character created:", character.id);
console.log(character);

File: examples/equos-python-integration/src/python_integration/create_character.py
import os

from equos.models import CreateEquosCharacterRequest

from python_integration.client import get_client


def create_character() -> None:
    client = get_client()

    brain_id = os.getenv("EQUOS_BRAIN_ID")
    if not brain_id:
        raise RuntimeError("EQUOS_BRAIN_ID is not set")

    character = client.characters.create(
        CreateEquosCharacterRequest(
            name="French Translator",
            livekit_identity=f"french-translator-{os.getpid()}",
            brain_id=brain_id,
            face_id=os.getenv("EQUOS_FACE_ID"),
            voice_id=os.getenv("EQUOS_VOICE_ID"),
            search=False,
        )
    )
    print("Character created:", character.id)


if __name__ == "__main__":
    create_character()
